### ISSUE TITLE
openTo parameter in file and image TVs aren't escaped

### ISSUE THREAD

**Issue by pdincubus** [modxbot/migrate#9526]
pdincubus created Redmine issue ID 9526

We've just happened across the following problem on a client site, where file pickers for images/files were not appearing after a save.

In manager/templates/default/element/tv/renders/input/file.tpl line 42, currently it's:

```
    {if $params.openTo},openTo: '{$params.openTo}'{/if}
```

It should be:

```
    {if $params.openTo},openTo: '{$params.openTo|replace:"'":"\\'"}'{/if}
```

There is the same issue on the image tpl, manager/templates/default/element/tv/renders/input/image.tpl line 40. Same fix for it.

Issue appeared initially on 2.1.3 but is still present on 2.2.6

**Comment by pdincubus** [comment]
pdincubus submitted:

Forgot to add - the reason this cropped up was due to client having a folder name under the assets directory which contained a ' character.

**Comment by pdincubus** [comment]
pdincubus submitted:

Pull request on GH: https://github.com/modxcms/revolution/pull/615

**Comment by pdincubus** [comment]
pdincubus submitted:

Done :)

**Comment by SnowCreative** [comment]
rainbowtiger submitted:

The same problem occurs if an ampersand is in the folder name.

**Comment by opengeek** [comment]
opengeek submitted:

Yeah is seems like this needs a more generic solution. It seems like maybe something is being used improperly here WRT to URL encoding/decoding.

Deferring this to 2.3.0 so we do not hold up critical patches in 2.2.10.

**Comment by opengeek** [comment]
opengeek submitted:

This is fixed in develop (for 2.3.0) and release-2.2 (for 2.2.11)
